app/src/main/kotlin/com/wire/android/ui/calling/initiating/InitiatingCallViewModel.kt

```diff
@@ -35,39 +32,58 @@ class InitiatingCallViewModel @Inject constructor(
     private val conversationDetails: ObserveConversationDetailsUseCase,
     private val allCalls: GetAllCallsUseCase,
     private val startCall: StartCallUseCase,
-    private val endCall: EndCallUseCase,
     private val callRinger: CallRinger
 ) : ViewModel() {
 
-    var callInitiatedState by mutableStateOf(InitiatingCallState())
-        private set
-
     val conversationId: QualifiedID = savedStateHandle
         .get<String>(EXTRA_CONVERSATION_ID)!!
         .parseIntoQualifiedID()
 
+    val conversationType: Deferred<ConversationType> = retrieveConversationTypeAsync()
+
     init {
         viewModelScope.launch {
-            launch { initiateCall() }
-            launch { initializeScreenState() }
-            launch { observeStartedCall() }
+            val job = launch { initiateCall() }
+            job.join()
+            observeStartedCall()
```

Comment:
not 10 0% but using job.join here is the same as :

viewModelScope.launch{
initiateCall()
observeStartedCall()
}